src/sentry/static/sentry/app/components/events/interfaces/spans/index.tsx

```diff
@@ -1,62 +1,170 @@
 import React from 'react';
 import styled from '@emotion/styled';
 import PropTypes from 'prop-types';
+import * as ReactRouter from 'react-router';
 
 import {t} from 'app/locale';
 import SearchBar from 'app/components/searchBar';
 import SentryTypes from 'app/sentryTypes';
 import {Panel} from 'app/components/panels';
 import space from 'app/styles/space';
 import EventView from 'app/utils/discover/eventView';
+import EventsV2 from 'app/utils/discover/eventsv2';
+import {stringifyQueryObject, QueryResults} from 'app/utils/tokenizeSearch';
+import AlertMessage from 'app/components/alertMessage';
+import {TableData} from 'app/views/eventsV2/table/types';
 
-import {SentryTransactionEvent} from './types';
+import {SentryTransactionEvent, ParsedTraceType} from './types';
+import {parseTrace, getTraceDateTimeRange} from './utils';
 import TraceView from './traceView';
 
-type PropType = {
+type Props = {
   orgId: string;
   event: SentryTransactionEvent;
   eventView: EventView;
-};
+} & ReactRouter.WithRouterProps;
 
 type State = {
+  parsedTrace: ParsedTraceType;
   searchQuery: string | undefined;
 };
 
-class SpansInterface extends React.Component<PropType, State> {
+class SpansInterface extends React.Component<Props, State> {
   static propTypes = {
     event: SentryTypes.Event.isRequired,
     orgId: PropTypes.string.isRequired,
   };
 
   state: State = {
     searchQuery: undefined,
+    parsedTrace: parseTrace(this.props.event),
   };
 
+  static getDerivedStateFromProps(props: Props, state: State): State {
+    return {
+      ...state,
+      parsedTrace: parseTrace(props.event),
+    };
+  }
+
   handleSpanFilter = (searchQuery: string) => {
     this.setState({
       searchQuery: searchQuery || undefined,
     });
   };
 
+  renderTraceErrorsAlert({
+    isLoading,
+    numOfErrors,
+  }: {
+    isLoading: boolean;
+    numOfErrors: number;
+  }) {
+    if (isLoading) {
+      return null;
+    }
+
+    if (numOfErrors === 0) {
+      return null;
+    }
+
+    const label =
+      numOfErrors > 1
+        ? t(
+            `There are %d error events associated with this transaction event.`,
+            numOfErrors
+          )
+        : t(`There is an error event associated with this transaction event.`);
+
+    return (
+      <AlertMessageContainer>
+        <AlertMessage
+          alert={{
+            id: 'id',
```

Comment:
Why `id` as the value? Could we do something like `transaction-alert`?

Reply:
done in https://github.com/getsentry/sentry/pull/18025/commits/f1b22473e0a39fce39fbe90f60a7ce1e371c31d2